group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -156,15 +165,16 @@ public CoordinatorRuntime<S, U> build() {
                 loader,
                 coordinatorShardBuilderSupplier,
                 time,
-                timer
+                timer,
+                runtimeMetrics
             );
         }
     }
 
     /**
      * The various state that a coordinator for a partition can be in.
      */
-    enum CoordinatorState {
+    public enum CoordinatorState {
```

Comment:
I don't immediately see an issue with this, but was there a reason this was not public before?

Reply:
it was only used within the coordinator runtime. now that we have some logic in the metrics package to handle state changes, we made it public.